### PR TITLE
Explode detailed tactic examples

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation

This PR is a follow-up of discussions with @jfehrle in https://github.com/coq/coq/pull/13892#issuecomment-826671536 and https://github.com/coq/coq/pull/14179#issuecomment-850875299 on removing the detailed tactic examples chapter and moving the examples closer to the tactic definitions.

This PR implements the plan while preserving the git history. It should be merged before the 8.14 branching point, because it won't be backportable.

### CODE REVIEW

**Comment by jfehrle** [comment]
Did you miss my comment "I think [the dependent induction example] should go into an appendix" or perhaps made a mistake?

**Comment by Zimmi48** [comment]
> Did you miss my comment "I think [the dependent induction example] should go into an appendix" or perhaps made a mistake?

No, I didn't miss your comment, on the contrary. That's just how I interpreted what you meant by "an appendix". I put the example toward the end to not break the flow of the chapter, instead of putting it close to the tactic definitions. And I named the subsection Appendix.

If you meant "create a whole new top-level chapter in the Appendix part of the refman just for this example", then I think:
- this is too early with so little to put in it;
- we would just reproduce the idea of a detailed tactic examples chapter that I wanted to move away from;
- we would also reproduce the idea of an appendix with random content that we used to have (it was called Addendum) and moved away from when restructuring the refman.

If your idea was something entirely different from the two proposals above, then please clarify.

File: doc/sphinx/proofs/writing-proofs/inductive.rst [hunk 1]
```diff
@@ -1401,4 +1401,305 @@ Generation of inversion principles with ``Derive`` ``Inversion``
...
+
+   dependent destruction v.
+
+In this case, the ``v`` variable can be replaced in the goal by the
+generalized hypothesis only when it has a type of the form ``vector (S n)``,
+that is only in the second case of the destruct. The first one is
+dismissed because ``S n <> 0``.
+
+
+A larger example
```
Comment (jfehrle) [inline doc/sphinx/proofs/writing-proofs/inductive.rst:1549]: For later: would like a better subtitle here.

File: doc/sphinx/proofs/writing-proofs/inductive.rst [hunk 2]
```diff
@@ -1401,4 +1401,305 @@ Generation of inversion principles with ``Derive`` ``Inversion``
 
     inversion H using leminv.
 
+.. _dependent-induction:
+
+Appendix: examples with dependent destruction / dependent induction
```
Comment (jfehrle) [inline doc/sphinx/proofs/writing-proofs/inductive.rst:1406]:
```suggestion
Examples of dependent destruction / dependent induction
```

**Review by jfehrle — APPROVED** [review]
Ah, appendix means "a section or table of additional matter _at the end_ of a book or document" (Oxford) or "supplementary material _usually attached at the end_ of a piece of writing" (Merriam-Webster).  But I think your idea will work and perhaps is better.

Perhaps you would tweak the section title, then I'll merge?
Only after that I can submit a PR renaming the file.

**Comment by Zimmi48** [comment]
OK, I've applied your suggestion.

**Comment by jfehrle** [comment]
Copacetic.  @coqbot: merge now

**Comment by jfehrle** [comment]
Oddly the PR says that `detailed_tactic_examples.rst` was deleted, but when I pull master and rebase, it's still there as a zero-length file and I get this error from Sphinx:

```
checking consistency...
/home/proj/coq2/doc/sphinx/proof-engine/detailed-tactic-examples.rst: WARNING: document isn't included in any toctree
```

But everything in your earlier build succeeded.

**Comment by Zimmi48** [comment]
I don't have this file on `master`. Maybe you had a change in this file and when rebasing you removed its content without removing the file, or there is a bug in git which keeps the file after the rebase? Anyway, I guess that a `git rm` should suffice to solve this issue.

**Pull request merged**